File: server/src/modules/notification/model/notification.model.js
const NotificationModel = {
  table: "notifications",

  field: {
    notificationId: "notification_id",
    userId: "user_id",
    title: "title",
    content: "content",
    isRead: "is_read",
    type: "type",
    createdAt: "created_at",
  },
};

module.exports = NotificationModel;

File: mobile/lib/features/notification/models/notification_model.dart
class AppNotificationModel {
  final String notificationId;
  final String title;
  final String content;
  final bool isRead;
  final String type;
  final DateTime createdAt;

  AppNotificationModel({
    required this.notificationId,
    required this.title,
    required this.content,
    required this.isRead,
    required this.type,
    required this.createdAt,
  });

  factory AppNotificationModel.fromJson(Map<String, dynamic> json) {
    return AppNotificationModel(
      notificationId: json['notification_id'] ?? json['id'] ?? '',
      title: json['title'] ?? 'Thông báo',
      content: json['content'] ?? json['body'] ?? '',
      isRead: json['is_read'] ?? json['isRead'] ?? false,
      type: json['type'] ?? 'SYSTEM',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'])
          : DateTime.now(),
    );
  }
}
